polygeist/tools/cgeist/Lib/utils.cc

```diff
@@ -12,82 +10,75 @@
 #include "clang-mlir.h"
 
 #include "mlir/IR/Builders.h"
-#include "mlir/IR/BuiltinOps.h"
 #include "mlir/IR/Operation.h"
 #include "mlir/IR/OperationSupport.h"
 #include "mlir/IR/Value.h"
 
 #include "llvm/ADT/ArrayRef.h"
-#include "llvm/ADT/STLExtras.h"
-#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/SmallVector.h"
 
-#include "mlir/Dialect/Linalg/IR/Linalg.h"
-#include "mlir/Dialect/MemRef/IR/MemRef.h"
+using namespace mlir;
 
-#include "clang/AST/Expr.h"
+namespace mlirclang {
```

Comment:
why do we not use `using namespace mlirclang;` in this case?

Reply:
https://stackoverflow.com/questions/59773471/scope-of-using-namespace-versus-using-namespace-closure

Reply:
Using `namespace MyNameSpace { ... }`  guarantees that any function definition in the namespace scope belong to the namespace.  The alternative is to declare each function with  MyNamepsace::MyFunc ...